parser/src/powdr.lalrpop

```diff
@@ -622,6 +633,22 @@ IfExpression: Box<Expression> = {
         <end:@R> => ctx.to_expr_with_source_ref(IfExpression{condition, body, else_body}, start, end),
 }
 
+StructExpression: Box<Expression> = {
+    <start:@L> <name:Identifier> "with" "{" <fields:StructExpressionList> "}" <end:@R> => ctx.to_expr_with_source_ref(StructExpression{name, fields}, start, end),
```

Comment:
What is the obstacle that forces us to use `with`?

Reply:
The parser cannot distinguish between Matchs and Structs when finding the first {. And if we try to modify MatchExpression in any way, then Structs and IfExpression become ambiguous.


I tried some variations to try to fix it but I couldn't, so I decided to stop wasting time and use a temporary patch.

Reply:
This very weird because `if` and `match` use keywords in front of the identifiers...

Do you know if `S({...})` works?

Reply:
> This very weird because `if` and `match` use keywords in front of the identifiers...
> 
> Do you know if `S({...})` works?

Yes, it works. In fact, anything between the `Identifier` and `{` of the struct works, because it helps disambiguate with the MatchExpression.

I think the problem comes from the `<scrutinee:BoxedExpression>` of the match, which can also be a Struct, so the parser can't tell if the first `{` is a Struct in the scrutinee or the beginning of the match branches.

We stick with S({...}) ?

Reply:
Yes, let's use `S({...})` for now. The rust grammar distinguishes between `ExpressionWithBlock` and `ExpressionWithoutBlock`: https://doc.rust-lang.org/reference/expressions.html

And match is in a different category than struct. Maybe that would help?

Reply:
Ah no, haha and that is interesting: https://doc.rust-lang.org/reference/expressions/match-expr.html

the scrutinee in the rust grammar excludes structs :)